doc/sphinx/source/tutorials/run-fit.rst

```diff
@@ -0,0 +1,282 @@
+``n3fit`` faster.
+
+.. note::
+
+    Tensorflow only supports the installation via pip. Note, however, that the TensorFlow pip package has been known to break third party packages. Install it at your own risk. Only the conda tensorflow-eigen package is tested by our CI systems.``
+
+When you install the nnpdf conda package, you get the
+`tensorflow-eigen <https://anaconda.org/anaconda/tensorflow-eigen>`__
+package, which is not the default. This is due to a memory explosion
+found in some of the conda mkl builds.
+
+If you want to disable MKL without installing ``tensorflow-eigen`` you
+can always set the environment variable ``TF_DISABLE_MKL=1`` before
+running ``n3fit``. When running ``n3fit`` all versions of the package
+show similar performance.
+
+When using the MKL version of tensorflow you gain more control of the
+way Tensorflow will use the multithreading capabilities of the machine
+by using the following environment variables:
+
+.. code:: bash
+
+
+   KMP_BLOCKTIME=0
+   KMP_AFFINITY=granularity=fine,verbose,compact,1,0
+
+These are the best values found for ``n3fit`` when using the mkl version
+of Tensorflow from conda and were found for TF 2.1 as the default values
+were suboptimal. For a more detailed explanation on the effects of
+``KMP_AFFINITY`` on the performance of the code please see
+`here <https://software.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top/optimization-and-programming-guide/openmp-support/openmp-library-support/thread-affinity-interface-linux-and-windows.html>`__.
+
+By default, ``n3fit`` will try to use as many cores as possible, but
+this behaviour can be overriden from the runcard with the ``maxcores``
+parameter. In our tests the point of diminishing returns is found at
+``maxcores=4``.
+
+Note that everything stated above is machine dependent so the best
+parameters for you might be very different. When testing, it is useful
+to set the environmental variable ``KMP_SETTINGS`` to 1 to obtain
+detailed information about the current variables being used by OpenMP.
+
+Below we present a benchmark that have been run for the Global NNPDF 3.1
+case, as found in the example runcards
+`folder <https://github.com/NNPDF/nnpdf/tree/master/n3fit/runcards>`__.
+
+Settings of the benchmark: 
+
+* TF version: tensorflow-eigen from conda, TF 2.2
+* NNPDF commit:
+`f878fc95a4f32e8c3b4c454fc12d438cbb87ea80 <https://github.com/NNPDF/nnpdf/commit/f878fc95a4f32e8c3b4c454fc12d438cbb87ea80>`__
+* Number of epochs: 5000 
+* maxcores: 4 
+* no early stopping
+
+Hardware: 
+
+* Intel(R) Core(TM) i7-6700 CPU @ 4.00GHz 
+* 16 GB RAM 3000 MHz DDR4
+
+Timing for a fit:
+
+* Walltime: 397s 
+* CPUtime: 1729s
+
+Iterate the fit
+---------------
+
+It may be desirable to iterate a fit to achieve a higher degree of
+convergence/stability in the fit. To read more about this, see `How to
+run an iterated fit <run-iterated-fit>`__.
```

Comment:
```suggestion
convergence/stability in the fit. To read more about this, see :ref:`How to
run an iterated fit <run-iterated-fit>`.
```

I think internal links should be like this instead?

Reply:
There are a few more than that (sorry, it wasn't clear)

Reply:
uh I see this is creating a lot of broken links :(